cloudwatch/cloudwatch.go

```diff
@@ -472,17 +507,9 @@ func encodeLogKey(log *interface{}) ([]byte, error) {
 
 // Flush sends the current buffer of records (for the stream that corresponds with the given tag)
 func (output *OutputPlugin) Flush(tag string) error {
-	if !output.logGroupCreated {
-		err := output.createLogGroup()
-		if err != nil {
-			return err
-		}
-		output.logGroupCreated = true
-	}
-
 	output.cleanUpExpiredLogStreams() // will periodically clean up, otherwise is no-op
 
-	stream, err := output.getLogStream(tag)
+	stream, err := output.getLogStream(tag, "")
```

Comment:
What do you think of simply duplicating the stream lookup by tag code here instead of passing an empty string in and relying on `getLogStream` method to handle it appropriately?

Reply:
I agree that this is not ideal but I'm not sure I understand what you mean exactly.

Reply:
Something like:
```go
name := output.getStreamName(tag)
stream, ok := output.streams[name]
if !ok {
	return fmt.Errorf("Failed to find stream with tag: %s", tag)
}
```

I believe this is the only logic we're relying on from `getLogStream` and this enables us to return a more informative error over `fmt.Errorf("groupName cannot be empty")` which could be returned from calling `getLogStream` with an empty group name.

Reply:
You're right. Done. :)